js/app.js

```diff
@@ -3,31 +3,110 @@ import { getCardComponent, createNewCardComponent } from './ui/card.js';
 import { addCard } from './data/actions.js';
 
 document.addEventListener('DOMContentLoaded', () => {
-  // Example of how we can create app state responsible for holding data
+  // display cards
+  // temporary show 2 fixed cards
+
   let appState = createState();
 
-  // Example of how we can create UI component using reusable function
-  const newCardData = { front: 'Good morning', back: 'Dzień dobry' };
+  const cardData = { front: 'Good morning', back: 'Dzień dobry' };
+  const secondCardData = { front: 'bad morning', back: 'kiepski poranek' };
 
-  const card = getCardComponent(newCardData);
+  const card = getCardComponent(cardData);
+  const secondCard = getCardComponent(secondCardData);
 
-  // Example of how we can add card to our state
-  const updatedAppState = addCard(appState, newCardData);
-  appState = updatedAppState;
-  card.getElementsByClassName('text').innerHTML = newCardData.front;
+  let newCardData = { front: '', back: '' };
 
   const cardList = document.getElementById('card-list');
 
-  // Example of how to display created card in our UI
-  // TODO - delay text change to mach flipping
-  cardList.append(card);
+  const fillCardList = () => {
+    while (cardList.firstChild) cardList.firstChild.remove();
+
+    cardList.append(card);
+    cardList.append(secondCard);
+  };
+
+  // add new card
+
+  const addNewCardButton = document.getElementById('add-card-button');
+
+  const addACard = () => {
+    const addNewCard = createNewCardComponent();
+    newCardData = { front: '', back: '' };
+
+    while (cardList.firstChild) cardList.firstChild.remove();
+    cardList.append(addNewCard);
+    const leftButton = addNewCard.querySelector('.btn-left');
+    const rightButton = addNewCard.querySelector('.btn-right');
+
+    const cancelButton = () => {
+      fillCardList();
+      if (!newCardData.front) addNewCard.querySelector('.new-value').value = '';
+    };
+
+    const nextButton = () => {
```

Comment:
**nitpick:** I think these methods (`nextButton, backButton, saveButton`) should be placed inside `createNewCardComponent` which creates UI for this card with these methods. Let's split what `addCard` should do (you can think of it as of splitting your code into smaller pieces):

1. It should initialize new card with its handlers using `createNewCardComponent()`
2. It should append this created card to the list
3. It should run `updateCounter`

Basically that's all. Each of the step could be reusable function

Reply:
Ahh, now I see that `createNewCardComponent` is your added card component. I suggest you to split these to as 2 different files, so you have your `app.js` clean. Think of it as one file should do one thing.